Platform/DeprecationWarner.swift

```diff
@@ -0,0 +1,48 @@
+//
+//  DeprecationWarner.swift
+//  Rx
+//
+//  Created by Shai Mishali on 1/9/18.
+//  Copyright © 2018 Krunoslav Zaher. All rights reserved.
+//
+
+import Foundation
+
+#if DEBUG
+    class DeprecationWarner {
+        private static var warned = [Kind]()
+        private static var _lock = NSRecursiveLock()
+        
+        static func warnIfNeeded(_ kind: Kind) {
+            _lock.lock(); defer { _lock.unlock() }
+            guard !warned.contains(kind) else { return }
+            
+            warned.append(kind)
+            print("ℹ️ [DEPRECATED] \(kind.message)")
+        }
+    }
+    
+    extension DeprecationWarner {
+        enum Kind {
+            case variable
+            case globalTestFunction(String)
+            
+            var message: String {
```

Comment:
I think you mean Equatable, not Hashable

And the problem is, I think this means we can't use an associated enum value (like I do for `globalTestFunction`). 

That would mean I'd have to do `globalTestFunctionNext, globalTestFunctionError, globalTestFunctionCompleted`, which is fine. (It would work with either Int or String or a regular Enum in this case, the problem is the associated value). 

Do you think I should just not have an associated values and opt for many enum cases ?

Reply:
This is what I'm referring to 

https://github.com/ReactiveX/RxSwift/pull/1546/files#diff-cc6f37b52e20a7eddf13182cba712c16R63

`DeprecationWarner.warnIfNeeded(.globalTestFunction("error"))`

vs. 
`DeprecationWarner.warnIfNeeded(.globalTestFunctionError)`